src/puyapy/awst_build/contract.py

```diff
@@ -421,84 +430,288 @@ def visit_type_alias_stmt(self, stmt: mypy.nodes.TypeAliasStmt) -> None:
         self._unsupported_stmt("type", stmt)
 
 
-def _gather_bases(contract_type: pytypes.ContractType) -> list[ContractReference]:
+class _UserContractBase(ContractFragmentBase, abc.ABC):
+    @property
+    @abc.abstractmethod
+    def options(self) -> ContractClassOptions | None: ...
+
+
+@attrs.frozen
+class _StaticContractBase(_UserContractBase):
+    id: ContractReference
+    methods_: dict[str, ContractFragmentMethod]
+    mro: Sequence[ContractFragmentBase]
+    symbols: dict[str, pytypes.PyType]
+    options: None = None
+
+    @typing.override
+    def resolve_method(
+        self, name: str, *, include_inherited: bool = True
+    ) -> ContractFragmentMethod | None:
+        return self.methods_.get(name)
+
+    @typing.override
+    def methods(
+        self, *, include_inherited: bool = True, include_overridden: bool = False
+    ) -> Iterator[ContractFragmentMethod]:
+        yield from self.methods_.values()
+
+    @typing.override
+    def resolve_storage(
+        self, name: str, *, include_inherited: bool = True
+    ) -> ContractFragmentStorage | None:
+        return None
+
+    @typing.override
+    def state(self, *, include_inherited: bool = True) -> Iterator[ContractFragmentStorage]:
+        yield from ()
+
+
+@attrs.frozen(kw_only=True)
+class _ContractFragment(_UserContractBase):
+    id: ContractReference
+    source_location: SourceLocation
+    pytype: pytypes.ContractType
+    mro: Sequence[_UserContractBase]
+    is_abstract: bool
+    options: ContractClassOptions
+    docstring: str | None
+    _methods: dict[str, ContractFragmentMethod] = attrs.field(factory=dict, init=False)
+    _state_defs: dict[str, ContractFragmentStorage] = attrs.field(factory=dict, init=False)
+    symbols: dict[str, pytypes.PyType | None] = attrs.field(factory=dict, init=False)
```

Comment:
Need to handle None?

Reply:
The None is to prevent going through to a base class if a symbol is overridden and type inference fails, so it's fine to treat None and not-found similarly in most cases, but this did make me realise that the super resolution was not doing so, so that's fixed now.